xapian-applications/omega/postprocess.py

```diff
@@ -0,0 +1,118 @@
+# Postprocess click data files for obtaining the final clickstream log file
+# and generate Query file for Xapian Letor module from it.
+#
+# Copyright (C) 2017 Vivek Pal
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+from ast import literal_eval
+from sys import argv
+
+import csv
+
+script_name, search_log, clicks_log = argv
+
+try:
+    with open(search_log, 'r') as s:
+        with open('log/clickstream.log', 'w+') as f:
+            reader = csv.reader(s, delimiter = '\t')
+            writer = csv.writer(f, delimiter = '\t', quoting = csv.QUOTE_MINIMAL)
+
+            for row in reader:
+                # Skip rows with empty Query string
+                if row[1] == '':
+                    continue
+                # Remove an extra ',' character at the end of the Hitlist string
+                hits = row[2]
+                hits = hits[:-2]
```

Comment:
Oh, seems like a typo. Should've been `1` here instead of `2`. I think it will not be required if `,` can be avoided by making use of `$hit` and `$sub{$last,1}` as you pointed out in a comment below. I'll go with that.